### DISCUSSION TITLE
Sampling request with ingress

### DISCUSSION THREAD

**Discussion by jdespinoza** [nginx/kubernetes-ingress#2614] [Q&A] [answered]
My log file is getting more data than usual so I need to do 1 of the below options:

1 - Sampling request, 1%
2 - Log only errors

What do you recommend me to follow? I already try the first option but I didn't get at all how it works and where do I need to setup the sampling request

**Comment by brianehlert** [comment]
What all information are you including in your logs? / what is your logging configuration?

Do you have access logs enabled?  
Because that can be a relatively high volume as it would be logging all requests if NGINX is configured as an API Gateway.

Also, there is this: https://docs.nginx.com/nginx-ingress-controller/configuration/global-configuration/configmap-resource/#logging

**Reply by jdespinoza** [reply]
This is my configmap.yaml

---

kind: ConfigMap

apiVersion: v1

metadata:

  name: nginx-configuration

  namespace: ingress-nginx

  labels:

    [app.kubernetes.io/name](http://app.kubernetes.io/name): ingress-nginx

    [app.kubernetes.io/part-of](http://app.kubernetes.io/part-of): ingress-nginx

data:

  worker-processes: "2"

  proxy_connect_timeout: "3s"

  proxy_read_timeout: "3s"

  proxy_send_timeout: "3s"

  disable-ipv6: "true"

  ssl-redirect: "false"

  ssl-protocols: "TLSv1 TLSv1.1 TLSv1.2 TLSv1.3"

  http-redirect-code: "301"

  hsts: "false"

  server-tokens: "false"

---

kind: ConfigMap

apiVersion: v1

metadata:

  name: tcp-services

  namespace: ingress-nginx

  labels:

    [app.kubernetes.io/name](http://app.kubernetes.io/name): ingress-nginx

    [app.kubernetes.io/part-of](http://app.kubernetes.io/part-of): ingress-nginx

---

kind: ConfigMap

apiVersion: v1

metadata:

  name: udp-services

  namespace: ingress-nginx

  labels:

    [app.kubernetes.io/name](http://app.kubernetes.io/name): ingress-nginx

    [app.kubernetes.io/part-of](http://app.kubernetes.io/part-of): ingress-nginx

**Reply by jdespinoza** [reply]
@brianehlert any input?

**Comment by jasonwilliams14** [comment]
@jdespinoza do you have your deployment/daemonset file available that I could take a look at?
So you are getting to much data in your logs and you want to trim that down?

**Reply by jdespinoza** [reply]
@jasonwilliams14 here is my deployment, and yes, I am getting to much data in my logs. I need to reduce that amount of data. I was thinking on below options:

- Log a sample set of NGINX access log request (1% of all traffic) to still allow me to get a baseline and alert on spikes from it
- Log nginx errors only (503/504)

--------------------------

apiVersion: apps/v1

kind: Deployment

metadata:

  name: nginx-ingress-controller

  namespace: ingress-nginx

  labels:

    [app.kubernetes.io/name](http://app.kubernetes.io/name): ingress-nginx

    [app.kubernetes.io/part-of](http://app.kubernetes.io/part-of): ingress-nginx

spec:

  replicas: 1

  selector:

    matchLabels:

      [app.kubernetes.io/name](http://app.kubernetes.io/name): ingress-nginx

      [app.kubernetes.io/part-of](http://app.kubernetes.io/part-of): ingress-nginx

  template:

    metadata:

      labels:

        [app.kubernetes.io/name](http://app.kubernetes.io/name): ingress-nginx

        [app.kubernetes.io/part-of](http://app.kubernetes.io/part-of): ingress-nginx

    spec:

      serviceAccountName: nginx-ingress-serviceaccount

      containers:

        - name: nginx-ingress-controller

          image: [us.gcr.io/k8s-artifacts-prod/ingress-nginx/controller:v1.1.1](http://us.gcr.io/k8s-artifacts-prod/ingress-nginx/controller:v1.1.1)

          args:

            - /nginx-ingress-controller

            - --configmap=$(POD_NAMESPACE)/nginx-configuration

            - --tcp-services-configmap=$(POD_NAMESPACE)/tcp-services

            - --udp-services-configmap=$(POD_NAMESPACE)/udp-services

            - --publish-service=$(POD_NAMESPACE)/ingress-nginx

            - --annotations-prefix=[nginx.ingress.kubernetes.io](http://nginx.ingress.kubernetes.io/)

            - --default-ssl-certificate=ingress-nginx/nginx-ingress

            - --enable-ssl-chain-completion=true

          securityContext:

            capabilities:

              drop:

                - ALL

              add:

                - NET_BIND_SERVICE

            runAsUser: 101

          env:

            - name: POD_NAME

              valueFrom:

                fieldRef:

                  fieldPath: [metadata.name](http://metadata.name/)

            - name: POD_NAMESPACE

              valueFrom:

                fieldRef:

                  fieldPath: metadata.namespace

          resources:

            requests:

              cpu: 750m

              memory: 750Mi

            limits:

              cpu: 1000m

              memory: 2000Mi

          ports:

            - name: http

              containerPort: 80

            - name: https

              containerPort: 443

          livenessProbe:

            failureThreshold: 3

            httpGet:

              path: /healthz

              port: 10254

              scheme: HTTP

            initialDelaySeconds: 10

            periodSeconds: 10

            successThreshold: 1

            timeoutSeconds: 10

          readinessProbe:

            failureThreshold: 3

            httpGet:

              path: /healthz

              port: 10254

              scheme: HTTP

            periodSeconds: 10

            successThreshold: 1

            timeoutSeconds: 10

**Comment by jasonwilliams14** [comment, marked as answer]
@jdespinoza Ahh, it looks like you are using a different NGINX Ingress Controller. From the above, looks like you are using the community version of Ingress, located here:

https://kubernetes.github.io/ingress-nginx/

HTH
